common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -74,44 +74,37 @@ sealed abstract class CodeExec[T <% SizeConversion](kind: String) extends Exec(k
     val pull = false
 
     // Whether the code is stored in a text-readable or binary format.
-    def binary: Boolean = false
-
-    override def size = code.sizeInBytes
-}
-
-sealed abstract class NodeJSAbstractExec(kind: String) extends CodeExec[String](kind) {
-    val main: Option[String]
-
-    override val entryPoint = main
     // the binary bit may be read from the database but currently it is always computed
     // when the "code" is moved to an attachment this may get changed to avoid recomputing
     // the binary property
-    override lazy val binary = Exec.isBinaryCode(code)
+    lazy val binary = {
+        code match {
+            case s: String => Exec.isBinaryCode(s)
+            case _         => false
+        }
+    }
+
+    override def size = code.sizeInBytes
 }
 
-protected[core] case class NodeJSExec(code: String, main: Option[String]) extends NodeJSAbstractExec(Exec.NODEJS)
-protected[core] case class NodeJS6Exec(code: String, main: Option[String]) extends NodeJSAbstractExec(Exec.NODEJS6)
+protected[core] abstract class CodeExecAsString(kind: String) extends CodeExec[String](kind)
```

Comment:
See WhiskAction for the reason - wanted to avoid a typeable or typecase when matching on the exec type since it's generic.